File: typescript/binary_tree_level_order_traversal.js
function levelOrder(root) {
    const q = [root];
    const res = [];
    while (q.length > 0) {
        const level = [];
        const qSize = q.length;
        for (let i = 0; i < qSize; i++) {
            const curr = q.shift();
            if (curr) {
                level.push(curr.val);
                q.push(curr.left);
                q.push(curr.right);
            }
        }
        if (level.length)
            res.push(level);
    }
    return res;
}
;

File: python/problems/binary_tree_level_order_traversal.py
from typing import Optional, List
from collections import deque

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        q = deque()
        q.append(root)

        while q:
            level = []
            for i in range(len(q)):
                curr = q.popleft()
                if curr:
                    level.append(curr)
                    q.append(curr.left)
                    q.append(curr.right)
            if level:
                ans.append(level)

        return ans 
